contracts/contracts/validator-registry/VanillaRegistry.sol

```diff
@@ -330,16 +343,18 @@ contract VanillaRegistry is IVanillaRegistry, VanillaRegistryStorage,
             require(_isUnstaking(pubKey), IVanillaRegistry.MustUnstakeToWithdraw());
             require(block.number > validator.unstakeOccurrence.blockHeight + unstakePeriodBlocks,
                 IVanillaRegistry.WithdrawingTooSoon());
+            require(validator.withdrawalAddress == expectedWithdrawalAddress,
+                IVanillaRegistry.WithdrawalAddressMismatch(validator.withdrawalAddress, expectedWithdrawalAddress));
             uint256 balance = validator.balance;
-            require(balance != 0, IVanillaRegistry.NothingToWithdraw());
```

Comment:
This was removed s.t. staking records can still be deleted even if all funds were slashed